### DISCUSSION TITLE
Compound unique identifier and nested update

### DISCUSSION THREAD

**Discussion by jdmoliner** [prisma/prisma#11846] [Q&A] [answered]
## Problem

I have a structure like the following. A user can have several subscriptions depending on the sport. The relationship between the `subscription` and the `user` is made with the user's id.

```prisma
enum Sport {
  BASKETBALL
  FOOTBALL
  HOCKEY
}

model User {
  id                       Int         @id  
  email                 String      @unique
  password           String  
  name                 String   
  subscription       Subscription[] 
}

model Subscription {
  id                     Int         @id @default(autoincrement())
  userId              Int
  user                 User        @relation(fields: [userId], references: [id])  
  sport               Sport
  status              String

  @@unique([userId, sport])
}

```
Then I need to do an update of `user` knowing only the `email` field
```ts
await prisma.user.update({
    where: {
      email: 'test@test.com'
    },
    data: {
      subscription: {
        update: {
          where: {
            userId_sport: {
              // userId: (here I don't know the user id)
              sport: 'FOOTBALL'   
            }                     
          },
          data: {
            status: 'active'
          }
        }
      }
    }
  })
```
The problem is that since in this case I don't know the `userId` I have to do something like this:

```ts
const user = await prisma.user.findUnique({
    where: { email: 'test@test.com' }
 })

await prisma.subscription.update({
    where: {      
      userId_sport: {
        userId: user.id,
        sport: 'FOOTBALL'
      }
    },
    data: {
       status: 'active'
    }
  })
```
Is there a way to do this in a single step?

**Comment by austincrim** [comment, marked as answer]
Hey @jdmoliner 👋,

This is not possible unless you change your schema to include the user's email in a `@@unique`` index in the `Subscription` model. Without this, we can't guarantee that we are updating a unique subscription row.

Let me know if I can clarify anything!

**Reply by JuanGalilea** [reply]
Shouldn't be the case that the only fields needed to be provided were those not known at the moment of query execution?
Since we are coming to `Subscription` through `User`, we clearly know `userId` at the moment of execution, and only need `sport` to access a unique field.

In fact, i'd say the syntax should be something along the lines of:
```
await prisma.user.update({
    where: {
      email: 'test@test.com'
    },
    data: {
      subscription: {
        update: {
          where: {
            sport: 'FOOTBALL'                   
          },
          data: {
            status: 'active'
          }
        }
      }
    }
  })
```

Shouldn't this exist in prisma?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
